packages/devtools_app/lib/src/shared/log_storage.dart

```diff
@@ -45,7 +45,8 @@ class LogStorage {
             if (e.stackTrace != null) 'stackTrace': e.stackTrace.toString(),
           }),
         )
-        .join('\n');
+        .map((line) => line + '\n') // Each entry needs to be on it's own line
+        .join();
```

Comment:
what about adding an extension method `joinWithTrailing(Pattern pattern)` that calls join and then appends one additional pattern at the end? We have `StringExtension` in utils.dart you can add this to.